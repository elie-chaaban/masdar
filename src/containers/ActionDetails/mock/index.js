export default [
  {
    id: 1,
    district_id: 1,
    module: 'ENERGY',
    section: 'ENERGY_DETAILS',
    portal_url: 'https://emm.nxncloud.io/pdt/smartpanels/viewer/districtview',
    portal_provider: 'TALISEN',
    token_required: 'N',
    uid: 'nxnadmin',
    pswd: 'NXNdemo_1',
    url_param: null,
    building_id: -99,
    expires_in: null,
    response: null,
    issued_date: null
  },
  {
    id: 4,
    district_id: 1,
    module: 'MOBILITY',
    section: 'MOBILITY_ASSETS',
    portal_url: 'http://82.212.107.181:8090/PenTrackDEMO',
    portal_provider: 'PENGUININ',
    token_required: 'N',
    uid: null,
    pswd: null,
    url_param: null,
    building_id: -99,
    expires_in: null,
    response: null,
    issued_date: null
  },
  {
    id: 8,
    district_id: 1,
    module: 'SECURITY',
    section: 'SECURITY_SURVEILLANCE',
    portal_url: 'http://nxn.3deye.me/',
    portal_provider: '3DEYE',
    token_required: 'Y',
    uid: 'ci@nxn.ae',
    pswd: 'ci@nxn.ae',
    url_param: null,
    building_id: -99,
    expires_in: 1800,
    response:
      '{"accessToken":"EriFbNLEnofNfuW38nCEt4_zPWidKTWEZWkhCet25_va9kPPv2YDn82-jGqaly3M9DwjUAlRVONPni07NCa-nuuyOvlgM6e5q3wLuyuGJ7P8MQwBWPlqVMPHkztcHlPDGoN13QXzFdOYFT5t0-MW-bcrzMQ_0hp3SEEHJv8Rm9QIx3yEml2bSWF9BEdRNMvicrWQqtGrjzHhthCPmAq8pu18h4A9Jjx5HNo3ZzNOGHTDn-zDaX-mxAtHk_Do0kRGzkl2Xewpr0BKAQFLAAbCfIsQE47foDInP7Ro6P13v6v56Y5HJAPygl3z7pLzS0lblGSvmU6yTTmxv1Kr7VRuOIVyb2INUDK9lsSexVmnwfkOLo6so8TFs2HzloB1-WsXjobuy1wdmfHIWKiJ4aTdAMyvq9M","tokenType":"bearer","expiresIn":1800,"refreshToken":"4713daa489aa4a90af04a5c989e46cc4","clientId":"ExternalApi","userName":"ci@nxn.ae","accessTokenIssued":"Wed, 02 Sep 2020 13:50:55 GMT","accessTokenExpires":"Wed, 02 Sep 2020 14:20:55 GMT","refreshTokenIssued":"Wed, 02 Sep 2020 13:50:55 GMT","refreshTokenExpires":"Wed, 23 Sep 2020 13:50:55 GMT"}',
    issued_date: '1599054655636'
  },
  {
    id: 12,
    district_id: 1,
    module: 'SECURITY',
    section: 'SECURITY_INCIDENTS',
    portal_url: 'https://security.nxncloud.io/vs4/situation/SituationDetails.jsp',
    portal_provider: 'VIDSYS',
    token_required: 'N',
    uid: null,
    pswd: null,
    url_param: null,
    building_id: -99,
    expires_in: null,
    response: null,
    issued_date: null
  }
];
