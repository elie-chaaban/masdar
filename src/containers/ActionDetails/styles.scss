@import '../../assets/styles/globals';

.dashboard-action-expandable {
  position: absolute;
  right: 0;
  top: 0;
  bottom: 0;
  width: 80vw;
  z-index: 100;
  .action-details-container {
    position: relative;
    height: 100vh;
    overflow: hidden;
    margin-left: 0;
    background-color: rgba(64, 64, 64, 1);
    z-index: 10001;

    .expandable-icon {
      position: fixed;
      top: 162px;
      transform: translateX(-25px);
    }
  }
}
