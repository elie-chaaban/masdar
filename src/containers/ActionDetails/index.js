import React, {useCallback, useEffect} from 'react';
import {useDispatch, useSelector, shallowEqual} from 'react-redux';
// import PeopleSearch from '../../components/Slider/Security/PeopleSearch';
import {toggleActionSlider} from '../../reduxStore/actions';
import './styles.scss';
import useFetch from '../../hooks';
import {SlideIn, Frame, ExpandableIcon} from '../../components/Utils';
import {getIFramesUrls} from '../../services/user';
import Search from '../../components/Slider/Security/Search';

const ActionDetails = () => {
  const dispatch = useDispatch();
  const map = useSelector((s) => s.map);
  let district,
    building = null;
  if (map) {
    district = map.district || null;
    building = map.building || null;
  }
  const buildingUrls = useFetch(getIFramesUrls, district, building);
  useEffect(() => {
    if (district || building) buildingUrls.fetch(true);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [district, building]);
  const credentials_3dEye = useSelector((s) => s.dashboard.credentials_3dEye, shallowEqual);
  const {open, action, data} = useSelector((s) => s.dashboard.actionSlider, shallowEqual);
  const triggerToggleActionSlider = () => dispatch(toggleActionSlider(action));
  const renderBody = useCallback(() => {
    if (action === 'Search') {
      return <Search />;
    } else {
      let iframe = null;
      if (building) {
        iframe = buildingUrls.response.districtPortals.find(
          (item) => item.section === data.toUpperCase() && item.module === action.toUpperCase() && item.buildingId === building
        );
      } else {
        iframe = buildingUrls.response.districtPortals.find(
          (item) => item.section === data.toUpperCase() && item.module === action.toUpperCase() && item.buildingId === null
        );
      }
      let url = iframe ? iframe.kpiConfiguration.apiUrl : '';
      let extraData = '',
        showLogoDiv = false;

      switch (data) {
        case 'SECURITY_SURVEILLANCE':
          extraData = `?access_token=${credentials_3dEye.accessToken}&refresh_token=${credentials_3dEye.refreshToken}`;
          // showLogoDiv = true;
          break;
        case 'SECURITY_INCIDENTS':
          extraData = `?situationID=${data.situation_id}&windowNumber=1`;
          break;
        case 'ENERGY_DETAILS':
          extraData = `?username=demoadmin&password=demo@NXN2018`;
          break;
        case 'ASSET_DETAILS':
          extraData = ``;
          break;
        default:
          break;
      }
      return <Frame source={url + extraData} showLogoDiv={showLogoDiv} />;
    }
  }, [data, credentials_3dEye, action, buildingUrls, building]);

  return (
    <SlideIn in={open && action !== 'Statistics'} className="dashboard-action-expandable">
      <div className="action-details-container">
        <ExpandableIcon onClick={triggerToggleActionSlider} active={open} inverted />
        {buildingUrls.response && action && action !== 'Statistics' && renderBody()}
      </div>
    </SlideIn>
  );
};
export default ActionDetails;
