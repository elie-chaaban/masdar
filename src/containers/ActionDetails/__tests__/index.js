import React from 'react';
import mockResponse from '../mock';
import {mockStore} from '../../../testUtils';
import {mount} from 'enzyme';
import ActionDetails from '..';
import {Provider} from 'react-redux';

jest.mock('../../../components/Slider/Security/PeopleSearch', () => ({
  __esModule: true,
  default: () => <div id="peopleSearch" />
}));
test('Action Details component People Search', () => {
  let store = mockStore({
    dashboard: {
      iFrames: mockResponse,
      credentials_3dEye: {accessToken: 'jfjfw134', refreshToken: 'refresh'},
      actionSlider: {data: null, action: 'Search', open: true}
    }
  });
  let dom = mount(
    <Provider store={store}>
      <ActionDetails />
    </Provider>
  );
  expect(dom).toMatchSnapshot();
  expect(dom.find('SlideIn').prop('in')).toBeTruthy();
  expect(dom.find('#peopleSearch')).toHaveLength(0);
  expect(dom.find('Frame')).toHaveLength(0);
  expect(dom.find('Memo(ExpandableIcon)').prop('active')).toBeTruthy();
  dom.find('Memo(ExpandableIcon)').props().onClick();
  expect(store.getActions()).toHaveLength(1);
  expect(store.getActions()[0]).toEqual({type: 'TOGGLE_ACTION_SLIDER', action: 'Search', data: null});
  dom.unmount();
  jest.clearAllMocks();
});
test('Action Details component energy', () => {
  let store = mockStore({
    dashboard: {
      iFrames: mockResponse,
      credentials_3dEye: {accessToken: 'jfjfw134', refreshToken: 'refresh'},
      actionSlider: {data: null, action: 'energy', open: true}
    }
  });
  let dom = mount(
    <Provider store={store}>
      <ActionDetails />
    </Provider>
  );
  // const url = mockResponse.find((item) => item.section === 'ENERGY' || item.module === 'ENERGY').portal_url;
  expect(dom.find('#peopleSearch')).toHaveLength(0);
  expect(dom.find('Frame')).toHaveLength(0);
  // expect(dom.find('Frame').prop('source')).toBe(url + '?username="demoadmin"&password="demo@NXN2018"');
  dom.unmount();
  jest.clearAllMocks();
});
test('Action Details component SECURITY INCIDENTS', () => {
  let store = mockStore({
    dashboard: {
      iFrames: mockResponse,
      credentials_3dEye: {accessToken: 'jfjfw134', refreshToken: 'refresh'},
      actionSlider: {data: {situation_id: '2'}, action: 'SECURITY_INCIDENTS', open: true}
    }
  });
  let dom = mount(
    <Provider store={store}>
      <ActionDetails />
    </Provider>
  );
  // const url = mockResponse.find((item) => item.section === 'SECURITY_INCIDENTS' || item.module === 'SECURITY_INCIDENTS').portal_url;
  expect(dom.find('#peopleSearch')).toHaveLength(0);
  expect(dom.find('Frame')).toHaveLength(0);
  // expect(dom.find('Frame').prop('source')).toBe(url + '?situationID=2&windowNumber=1');
  dom.unmount();
  jest.clearAllMocks();
});
test('Action Details component SECURITY SURVEILLANCE', () => {
  let store = mockStore({
    dashboard: {
      iFrames: mockResponse,
      credentials_3dEye: {accessToken: 'jfjfw134', refreshToken: 'refresh'},
      actionSlider: {data: {situation_id: '2'}, action: 'SECURITY_SURVEILLANCE', open: true}
    }
  });
  let dom = mount(
    <Provider store={store}>
      <ActionDetails />
    </Provider>
  );
  // const url = mockResponse.find((item) => item.section === 'SECURITY_SURVEILLANCE' || item.module === 'SECURITY_SURVEILLANCE').portal_url;
  expect(dom.find('#peopleSearch')).toHaveLength(0);
  expect(dom.find('Frame')).toHaveLength(0);
  // expect(dom.find('Frame').prop('source')).toBe(url + '?access_token=jfjfw134&refresh_token=refresh');
  dom.unmount();
  jest.clearAllMocks();
});
test('Action Details component Statistics', () => {
  let store = mockStore({
    dashboard: {
      iFrames: mockResponse,
      credentials_3dEye: {accessToken: 'jfjfw134', refreshToken: 'refresh'},
      actionSlider: {data: {situation_id: '2'}, action: 'Statistics', open: true}
    }
  });
  let dom = mount(
    <Provider store={store}>
      <ActionDetails />
    </Provider>
  );
  expect(dom.find('SlideIn').prop('in')).toBeFalsy();
  expect(dom.find('#peopleSearch')).toHaveLength(0);
  expect(dom.find('Frame')).toHaveLength(0);
  expect(dom.find('Memo(ExpandableIcon)')).toHaveLength(0);
  dom.unmount();
  jest.clearAllMocks();
});
