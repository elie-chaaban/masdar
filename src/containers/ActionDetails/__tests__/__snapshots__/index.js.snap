// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Details component People Search 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ActionDetails>
    <SlideIn
      className="dashboard-action-expandable"
      in={true}
    >
      <Animate
        animation="fadeRight"
        className="dashboard-action-expandable"
        enter={true}
        onAnimationEnd={null}
        options={
          Object {
            "duration": 500,
          }
        }
        unMountOnExit={true}
      >
        <div
          className="dashboard-action-expandable"
          onAnimationEnd={[Function]}
          style={
            Object {
              "animationDelay": "undefinedms",
              "animationDirection": undefined,
              "animationDuration": "500ms",
              "animationFillMode": undefined,
              "animationIterationCount": undefined,
              "animationName": "fadeRightIn",
              "animationPlayState": undefined,
              "animationTimingFunction": undefined,
            }
          }
        >
          <div
            className="action-details-container"
          >
            <Memo(ExpandableIcon)
              active={true}
              inverted={true}
              onClick={[Function]}
            >
              <Animate
                animation="fade"
                enter={true}
                onAnimationEnd={null}
                options={
                  Object {
                    "duration": 500,
                  }
                }
                unMountOnExit={true}
              >
                <div
                  onAnimationEnd={[Function]}
                  style={
                    Object {
                      "animationDelay": "undefinedms",
                      "animationDirection": undefined,
                      "animationDuration": "500ms",
                      "animationFillMode": undefined,
                      "animationIterationCount": undefined,
                      "animationName": "fadeIn",
                      "animationPlayState": undefined,
                      "animationTimingFunction": undefined,
                    }
                  }
                >
                  <Icon
                    className="expandable-icon clickable"
                    height="50"
                    name="ExpandableInverted"
                    onClick={[Function]}
                    width="25"
                  >
                    <ForwardRef(SvgExpandableInvertedIcon)
                      className="expandable-icon clickable"
                      height="50"
                      onClick={[Function]}
                      width="25"
                    >
                      <svg
                        className="expandable-icon clickable"
                        height="50"
                        onClick={[Function]}
                        width="25"
                      >
                        expandable-inverted-icon.svg
                      </svg>
                    </ForwardRef(SvgExpandableInvertedIcon)>
                  </Icon>
                </div>
              </Animate>
            </Memo(ExpandableIcon)>
          </div>
        </div>
      </Animate>
    </SlideIn>
  </ActionDetails>
</Provider>
`;